datasets/datasets.py

```diff
@@ -70,36 +71,38 @@ def create_dataset(file_object):
         file = file_object.file
         filename = file_object.filename
 
-    # if user does not select file, the browser also
-    # submits an empty part without filename
-    if filename == "":
-        raise BadRequest("No selected file.")
-
-    # generate a dataset name from filename
-    name = generate_name(filename)
-
-    try:
-        # reads file into a DataFrame
-        df = read_into_dataframe(file, filename)
-    except Exception:
-        # if read fails, then uploads raw file
-        file.seek(0, SEEK_SET)
-        save_dataset(name, file, metadata={"original-filename": filename})
-        return {"name": name, "filename": filename}
-
-    columns = df.columns.values.tolist()
-    featuretypes = infer_featuretypes(df)
-
-    metadata = {
-        "columns": columns,
-        "featuretypes": featuretypes,
-        "original-filename": filename,
-        "total": len(df.index),
-    }
-
     file.seek(0, SEEK_SET)
```

Comment:
Esse file.seek está aqui pois caso haja um erro de leitura o código precisa na próxima chamada direcionar o ponteiro de leitura para o início. As outras partes do código que possuem buffer.seek tem uma ideia parecida, que é, garantir que o ponteiro tenha voltado pra posição inicial.